core/src/main/scala/com/stripe/dagon/Dag.scala

```diff
@@ -126,6 +126,13 @@ sealed abstract class Dag[N[_]] { self =>
   }
 
   /**
+   * Apply a sequence of rules in order
+   * First apply one rule until it does not apply, then the next, etc..
+   */
+  def applySeq(rules: Seq[Rule[N]]): Dag[N] =
```

Comment:
want to call these `phases` instead of `rules` to sort of hint at the ordering part?